### DISCUSSION TITLE
How to authenticate other users

### DISCUSSION THREAD

**Discussion by akoushke** [hammem/monarchmoney#96] [Q&A] [answered]
Right now, this library lets users log in in two ways: by entering their username and password (interactive, non-interactive). I know it doesn't have OAuth, and I'm avoiding basic Authentication because it's not secure. I'm wondering if we can use this library to add features for other users, or if it's just for developers making their own tools for managing Monarch data. And if we do use basic authentication, how can we make sure it's safe? SSL might help a bit, but it's not perfect. Am I missing anything important? 

Note: I'm not trying to manage other users passwords also.

**Comment by hammem** [comment, marked as answer]
Thanks for the question, @akoushke !  In short, this library is not intended for use by a third-party to authenticate securely on another user's behalf to access and/or process their data.  Until Monarch enables its own developer API integration, like Google and other major platforms do, it won't be possible.

In limited circumstances, you may find it permissible to use Monarch's Financial Advisors feature, which gives a trusted third-party time-bound, read-only access to an account via a separate set of credentials.  This feature isn't all that well documented, but there is some details written for those acting as financial advisors: https://help.monarchmoney.com/hc/en-us/articles/11251085750292-Advisor-Program-Now-in-Beta.
